clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -992,6 +997,11 @@ private PartitionRecords parseCompletedFetch(CompletedFetch completedFetch) {
             } else if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
                 log.warn("Not authorized to read from topic {}.", tp.topic());
                 throw new TopicAuthorizationException(Collections.singleton(tp.topic()));
+            } else if (error == Errors.FENCED_LEADER_EPOCH) {
```

Comment:
We could probably add this to the errors on line 979 above. This is a "normal" error case, so I think we logging at debug level should be sufficient.